### DISCUSSION TITLE
Non-core block dependencies in themes and patterns

### DISCUSSION THREAD

**Discussion by luisherranz** [WordPress/gutenberg#35471] [Developer Experience]
Regarding non-core blocks in themes, I wonder:

- What are the plans for non-core block support in theme templates and patterns? Will they be allowed?
  - If so, is there a plan for some kind of dependency system, or how will that be left out for theme creators to figure out?
  - If not, are all the themes and patterns released by the community supposed to use only core blocks?

And the same about patterns:

- Are patterns with non-core blocks allowed in the Block Pattern Directory?
- Again: if so, how are they going to work? How a pattern will declare its dependencies?

**Comment by mtias** [comment]
The pattern directory already contemplates using blocks that are part of the block directory, given there are mechanisms for installing / activating those from the UI when the user inserts one. The flow is rudimentary, since the user cannot preview it in their own site before installing a missing block, but workable. The limitation of using blocks from the directory is because it deals with single-block plugins.

**Reply by luisherranz** [reply]
Thanks a lot for the response Matías, and sorry for my late answer. It makes total sense to manage this in the UI 🙂
